shared/bundle_analysis/report.py

```diff
@@ -34,54 +33,58 @@ def size(self):
         return self.asset.size
 
     def modules(self):
-        return (
-            self.db_session.query(models.Module)
-            .join(models.Module.chunks)
-            .join(models.Chunk.assets)
-            .filter(models.Asset.id == self.asset.id)
-            .all()
-        )
+        with models.get_db_session(self.db_path) as session:
```

Comment:
Will all of these open the connection, do the logic, then close the connection?

Reply:
Yes, the SQLAlchemy session object can act as a context manager.